pkg/cli/cmd/config/set_test.go

```diff
@@ -51,23 +109,23 @@ func TestSetCommand(t *testing.T) {
 				t.Errorf("Expected Short to be %q, got %q", tt.wantShort, cmd.Short)
 			}
 
-			// Verify flag exists and has correct default value
-			noPromptFlag := cmd.Flag("no-prompt")
-			if noPromptFlag == nil {
-				t.Fatal("Expected no-prompt flag to exist")
+			// Initialize the flags by parsing empty args
+			if err := cmd.ParseFlags([]string{}); err != nil {
+				t.Fatalf("Failed to parse flags: %v", err)
 			}
 
-			// Check default value
-			if noPromptFlag.Value.String() != "false" {
-				t.Errorf("Expected no-prompt flag to be false by default, got %s", noPromptFlag.Value.String())
+			// Set flags if provided
+			for flag, value := range tt.flags {
+				if err := cmd.Flags().Set(flag, value); err != nil {
+					t.Errorf("Failed to set %q flag: %v", flag, err)
+				}
 			}
 
-			// Test flag setting
-			if err := noPromptFlag.Value.Set("true"); err != nil {
-				t.Errorf("Failed to set no-prompt flag: %v", err)
-			}
-			if noPromptFlag.Value.String() != "true" {
-				t.Errorf("Expected no-prompt flag to be true after setting, got %s", noPromptFlag.Value.String())
+			// Verify prompting behavior
+			prompt := !cmd.Flags().Changed("host") && !cmd.Flags().Changed("token") &&
+				!cmd.Flags().Changed("api-path") && !cmd.Flags().Changed("insecure-skip-tls-verify")
```

Comment:
Same comment as above. BTW it feels a bit strange to me we duplicate code from the file we test in the test file. Maybe make that a separate function, but I really do not have strong opinion on that (feel free to keep as is).

Reply:
Yes, makes sense. I did the required changes.